codespell_lib/_codespell.py

```diff
@@ -602,7 +602,7 @@ def parse_file(filename, colors, summary, misspellings, exclude_lines,
             return bad_count
 
     for i, line in enumerate(lines):
-        if line in exclude_lines:
+        if line.rstrip() in exclude_lines:
```

Comment:
I've realised the issue. This will write back out. See the `-w` based tests (and extend them to prove it/stop us breaking it anyway).

An input of `abandonned\nAbandonned\r\nABANDONNED \nAbAnDoNnEd` will become `abandonned\nAbandonned\nABANDONNED\nAbAnDoNnEd` I think

Reply:
> An input of `abandonned\nAbandonned\r\nABANDONNED \nAbAnDoNnEd` will become `abandonned\nAbandonned\nABANDONNED\nAbAnDoNnEd` I think

It becomes `abandoned\nAbandoned\nABANDONED \nabandoned` on both codespell 2.0.0 and on the version with modifications from this PR.
I'm not quite sure what and why I should be extending tests here as the changes from this PR do not make any changes that would cause any regression. I feel like it would be better suited for a separate PR?

If you don't agree, I can make the changes here, just let me know *what test* I should be modifying and/or what kind of inputs/outputs should I be checking there.

Reply:
@peternewman do you think I could get your decision here so that I know if this is fine or if I should work on something before this will be able to move forward?

Reply:
ping @peternewman

Reply:
> > An input of `abandonned\nAbandonned\r\nABANDONNED \nAbAnDoNnEd` will become `abandonned\nAbandonned\nABANDONNED\nAbAnDoNnEd` I think
> 
> It becomes `abandoned\nAbandoned\nABANDONED \nabandoned` on both codespell 2.0.0 and on the version with modifications from this PR.

Odd, as we added the tests here in #2490 and they seem to behave okay, although admittedly they are a slightly different code path to the command line version. What OS were you testing on?

> I'm not quite sure what and why I should be extending tests here as the changes from this PR do not make any changes that would cause any regression. I feel like it would be better suited for a separate PR?

I believe the change you're making will introduce a bug, or perhaps simply further embed an existing bug, so I figure fixing the crucial underlying one first is probably a good idea.

> If you don't agree, I can make the changes here, just let me know _what test_ I should be modifying and/or what kind of inputs/outputs should I be checking there.

As mentioned, I think we've now done that in #2490 so when that's in, you can just update/rebase and as long as your code continue to pass we'll be all good.

Reply:
> What OS were you testing on?

Back then probably only on Windows 10.

> I believe the change you're making will introduce a bug, or perhaps simply further embed an existing bug, so I figure fixing the crucial underlying one first is probably a good idea.

That certainly makes sense, I am just not really sure what the issue is exactly. Hopefully, the tests added in #2490 will make it more clear whether there is any issue and if so, what that issue is, thanks for working on those tests. Once that PR gets merged, I'll rebase.
If any further work on the PR will be need, I might not get to it before Saturday but other than that, I'll be happy to make any necessary changes :+1:

Reply:
I rebased the PR on top of current `master` now that the tests got merged. Locally the tests ran fine, you'll have to approve the CI run though since I haven't had any PRs into this repository yet.

---

By the way, in case you don't know, the requirement for approval of workflows for first-time contributors who are NOT new to GitHub can be disabled to decrease the contribution barrier for first-time contributors, see:
https://twitter.com/JakubKuczys/status/1562660410108841986